plugin/neomake.vim

```diff
@@ -18,11 +18,17 @@ function! s:NeomakeCommand(file_mode, enabled_makers)
     endif
 endfunction
 
+function! s:NeomakeSh(...)
+    call neomake#Make({'sh_command': join(a:000, ' ')})
+endfunction
+
 command! -nargs=* -bang Neomake call s:NeomakeCommand('<bang>' !=# '!', [<f-args>])
 " These commands are available for clarity
 command! -nargs=* NeomakeProject Neomake! <args>
 command! -nargs=* NeomakeFile Neomake <args>
 
+command! -nargs=+ NeomakeSh call s:NeomakeSh(<f-args>)
```

Comment:
I think instead of `<f-args>` you want `<q-args>`, which will just deliver a single string, exactly as they entered it. Since there isn't a lot to the `s:NeomakeSh` function, you could replace this line with:

```
command! -nargs=+ NeomakeSh call neomake#Make({'sh_command': <q-args>})
```